examples/basic_msg_test/basic_msg_test.c

```diff
@@ -0,0 +1,194 @@
+#include <inttypes.h>
+#include <stdarg.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/queue.h>
+#include <unistd.h>
+
+#include <rte_common.h>
+#include <rte_ip.h>
+#include <rte_mbuf.h>
+#include <rte_mempool.h>
+#include <rte_ring.h>
+#include <rte_malloc.h>
+
+#include "onvm_nflib.h"
+#include "onvm_pkt_helper.h"
+
+#define NF_TAG "basic_msg_test"
+
+/* number of package between each print */
+static uint32_t print_delay = 1000000;
+
+static uint16_t destination;
+
+void
+nf_setup(struct onvm_nf_local_ctx *nf_local_ctx);
+
+void
+nf_msg_handler(void *msg_data, struct onvm_nf_local_ctx *nf_local_ctx);
+
+/*
+ * Print a usage message
+ */
+static void
+usage(const char *progname) {
+        printf("Usage:\n");
+        printf("%s [EAL args] -- [NF_LIB args] -- -p <print_delay>\n", progname);
+        printf("%s -F <CONFIG_FILE.json> [EAL args] -- [NF_LIB args] -- [NF args]\n\n", progname);
+        printf("Flags:\n");
+        printf(" - `-p <print_delay>`: number of packets between each print, e.g. `-p 1` prints every packets.\n");
+}
+
+/*
+ * Parse the application arguments.
+ */
+static int
+parse_app_args(int argc, char *argv[], const char *progname) {
+        int c;
+
+        while ((c = getopt(argc, argv, "p:")) != -1) {
+                switch (c) {
+                        case 'p':
+                                print_delay = strtoul(optarg, NULL, 10);
+                                RTE_LOG(INFO, APP, "print_delay = %d\n", print_delay);
+                                break;
+                        case '?':
+                                usage(progname);
+                                if (optopt == 'p')
+                                        RTE_LOG(INFO, APP, "Option -%c requires an argument.\n", optopt);
+                                else if (isprint(optopt))
+                                        RTE_LOG(INFO, APP, "Unknown option `-%c'.\n", optopt);
+                                else
+                                        RTE_LOG(INFO, APP, "Unknown option character `\\x%x'.\n", optopt);
+                                return -1;
+                        default:
+                                usage(progname);
+                                return -1;
+                }
+        }
+        return optind;
+}
+
+void
+nf_setup(__attribute__((unused)) struct onvm_nf_local_ctx *nf_local_ctx) {
+        struct rte_mempool *pktmbuf_pool;
+
+        pktmbuf_pool = rte_mempool_lookup(PKTMBUF_POOL_NAME);
+        if (pktmbuf_pool == NULL) {
+                onvm_nflib_stop(nf_local_ctx);
+                rte_exit(EXIT_FAILURE, "Cannot find mbuf pool!\n");
+        }
+
+        uint16_t address = 1;
+
+        int ret = onvm_nflib_send_msg_to_nf(address, NULL);
+        printf("%d\n", ret);
+        
+}
+
+void
+nf_msg_handler(void *msg_data, __attribute__((unused)) struct onvm_nf_local_ctx *nf_local_ctx){
+
+        uint16_t address = 1;
```

Comment:
Could you maybe make this command line configurable?

Reply:
Working on this now.

Reply:
for this particular NF let's not worry about making it a command line argument -- my eventual goal will be for this "test nf" to be a unit test framework to test a variety of different functions. I'll add more about this in my review.

Reply:
That is fair - as long as this NF always has a service ID of 1 it should work as expected. There will be problems if there are other NF's started before this one is, however.

Reply:
yeah good point.  Since the goal here is just to send packets to the NF itself, a better solution would be to fill in the destination ID based on this NF's service ID. That should be available in the `onvm_nf_local_ctx` struct. Then it will work regardless of what service ID we use when running the NF.

Reply:
Yea, that is a good approach

Reply:
I believe I have resolved this issue.